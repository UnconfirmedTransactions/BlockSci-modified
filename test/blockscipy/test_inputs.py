import pytest


def test_p2pkh_input_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2pkh-spend-1-tx"])
    input_regression(tx, regtest)


def test_p2sh_input_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2sh-spend-2-tx"])
    input_regression(tx, regtest)


@pytest.mark.btc
@pytest.mark.ltc
def test_p2wpkh_input_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2wpkh-spend-0-tx"])
    input_regression(tx, regtest)


@pytest.mark.btc
@pytest.mark.ltc
def test_p2wsh_input_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2wsh-spend-1-tx"])
    input_regression(tx, regtest)


def input_regression(tx, regtest):
    for inpt in tx.inputs:
        print(inpt.address, file=regtest)
        print(inpt.address_type, file=regtest)
        print(inpt.age, file=regtest)
        print(inpt.block, file=regtest)
        print(inpt.index, file=regtest)
        print(inpt.sequence_num, file=regtest)
        print(inpt.spent_output, file=regtest)
        print(inpt.spent_tx, file=regtest)
        print(inpt.spent_tx_index, file=regtest)
        print(inpt.tx, file=regtest)
        print(inpt.tx_index, file=regtest)
        print(inpt.value, file=regtest)


def test_input_output_mapping(chain):
    for block in chain:
        for tx in block:
            for inpt in tx.inputs:
                assert inpt == inpt.spent_output.spending_input
                assert tx == inpt.spent_output.spending_tx
