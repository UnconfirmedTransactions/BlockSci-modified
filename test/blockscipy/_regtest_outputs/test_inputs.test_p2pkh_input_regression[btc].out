PubkeyHashAddress(mjTkW3DjgyZck4KbiRusZsqTgaYTxdSz6z)
Pay to pubkey hash
110
Block(tx_count=3, height=112, header_hash=628da9df97ec8feb26bedfe08745be3eb94e322bbd25dac59c2efac07510fdf5, version=536870912, timestamp=1535761800, bits=545259519, nonce=2)
0
4294967295
TxOut(spending_tx_index=114, address=PubkeyHashAddress(mjTkW3DjgyZck4KbiRusZsqTgaYTxdSz6z), value=5000000000)
Tx(len(txins)=0, len(txouts)=2, size_bytes=144, block_height=2, tx_index=2)
2
Tx(len(txins)=1, len(txouts)=2, size_bytes=226, block_height=112, tx_index=114)
114
5000000000
